### PR TITLE
diamond: add new exercise

### PR DESCRIPTION
**Pull request by filiptoskovic**

### CODE REVIEW

File: exercises/diamond/uDiamondTest.pas [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+procedure TDiamondTest.Degenerate_case_with_a_single_A_row;
+begin
+  Expected := ['A'];
+  CompareArrays(Expected, TDiamond.Rows('A'));
+end;
+
+procedure TDiamondTest.Degenerate_case_with_no_row_containing_3_distinct_groups_of_spaces;
+begin
+  Expected := [ ' A ',
```
Comment (rpottsoh) [inline exercises/diamond/uDiamondTest.pas]: Very simple white space correction.  Space between `[` and `'` should be removed.

File: exercises/diamond/uDiamondTest.pas [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                'B B',
+                ' A '];
+  CompareArrays(Expected, TDiamond.Rows('B'));
+end;
+
+procedure TDiamondTest.Largest_possible_diamond;
+begin
+  Expected := [ '                         A                         ',
```
Comment (rpottsoh) [inline exercises/diamond/uDiamondTest.pas]: Same as previous comment.

File: exercises/diamond/uDiamondTest.pas [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+               'D     D',
+               ' C   C ',
+               '  B B  ',
+               '   A   '];
+  CompareArrays(Expected, TDiamond.Rows('D'));
+end;
+
+procedure TDiamondTest.Smallest_non_degenerate_case_with_odd_diamond_side_length;
+begin
+  Expected := ['  A  ',
```
Comment (rpottsoh) [inline exercises/diamond/uDiamondTest.pas]: This one is OK.

**Comment by rpottsoh** [comment]
Nice work @filiptoskovic.  Thanks for working on this.  I'll whip a config update.

**Comment by filiptoskovic** [comment]
I will have to use auto format option.

**Comment by filiptoskovic** [comment]
Diamond is hard? Difficulty 8? I think it is overrated.

**Pull request merged**